lib/rubocop/cop/performance/case_when_splat.rb

```diff
@@ -63,6 +63,11 @@ class CaseWhenSplat < Cop
         PERCENT_I = '%i'.freeze
         PERCENT_CAPITAL_I = '%I'.freeze
 
+        def initialize(*)
+          super
+          @reordered_splat_condition = false
+        end
```

Comment:
Soon, I would like to add hooks to all cops which will be called when beginning and ending inspection of a file.

Right now, each cop instance is only used for a single inspection run of a single file. To enhance performance, it would be better if cop instances can be reused. But we need a place to reinitialize state variables on each inspection run. The hook will provide a good place to do that.

After that, this override of `initialize` will not be needed.